blocks/library/image/block.js

```diff
@@ -220,10 +224,69 @@ class ImageBlock extends Component {
 								onChange={ this.updateImageURL }
 							/>
 						) }
+						<div className="blocks-image-dimensions">
+							<p className="blocks-image-dimensions__row">
+								{ __( 'Image Dimensions' ) }
+							</p>
+							<div className="blocks-image-dimensions__row">
+								<TextControl
+									type="number"
+									className="blocks-image-dimensions__width"
+									label={ __( 'Width' ) }
+									value={ width !== undefined ? width : '' }
+									placeholder={ selectedSize.width }
+									onChange={ ( value ) => {
```

Comment:
Minor: I know this is not the first usage of arrow functions as props here but since this is already a class, we could use bound methods instead.